includes/admin/reporting/class-base-logs-list-table.php

```diff
@@ -346,20 +353,47 @@ protected function get_query_args( $paginate = true ) {
 			if ( filter_var( $search, FILTER_VALIDATE_IP ) ) {
 				$retval['ip'] = $search;
 			} elseif ( is_email( $search ) ) {
-				$customer = edd_get_customer( $search );
-				if ( ! empty( $customer->id ) ) {
-					$retval['customer_id'] = $customer->id;
+				if ( 'api_requests' === $this->log_type ) {
+					// API requests are linked to user accounts, so we're checking user data here.
+					$user = get_user_by( 'email', $search );
+					if ( ! empty( $user->ID ) ) {
```

Comment:
I think that this and line 365 are exhibiting the same behavior for me, and returning all results on any log search when I input an invalid email address--because the user/customer ID is not added as a parameter to the `$retval` array, it's not factored in to the search results. This feels weird to me because if I enter an invalid email address, I would think I should get no results, but it looks like this behavior is the same in 2.x as well, so it's not a regression, just something I don't like?

The updated email search is working in the API Request logs now, as long as I put in a valid email address.

Reply:
Okay just pushed up a change that I believe should fix that.

This function is officially a janky piece of mess and may be worth refactoring at some point. But not now. :stuck_out_tongue: